Switching from ancestry to TOB-WGS branch

KatalinaBobowik opened the pull request:
The original branch is located [here](https://github.com/populationgenomics/ancestry/pull/182)

Commit:
Switching from ancestry to TOB-WGS branch

lgruen commented on scripts/vep/main.py:
```diff
@@ -0,0 +1,57 @@
+
+"""
+Use VEP using a dataproc cluster.
+Taken from Matt Welland's script, run_vep_help.py
```
Could you add a GitHub permalink for that instead?

KatalinaBobowik replied:
unfortunately I don't think that script exists anymore

lgruen replied:
Okay, maybe just remove that comment line then :).

KatalinaBobowik replied:
Okay, removed

lgruen commented on scripts/vep/main.py:
```diff
@@ -0,0 +1,57 @@
+    service_backend = hb.ServiceBackend(
+        billing_project=os.getenv('HAIL_BILLING_PROJECT'),
+        bucket=os.getenv('HAIL_BUCKET'),
+    )
```
Does that still work? Don't you need to use something like https://github.com/populationgenomics/analysis-runner/blob/main/examples/batch/hello.py#L25-L28 instead?

lgruen commented on scripts/vep/main.py:
```diff
@@ -0,0 +1,57 @@
+    )
+    job.cpu(2)
+    job.memory('standard')
+    job.storage('20G')
```
```suggestion
```

lgruen commented on scripts/vep/main.py:
```diff
@@ -0,0 +1,57 @@
+    job = dataproc.hail_dataproc_job(
+        batch=batch,
+        worker_machine_type='n1-highmem-8',
+        worker_boot_disk_size=200,
+        secondary_worker_boot_disk_size=200,
```
Do you need this?

KatalinaBobowik replied:
I'm not sure, I can test without

vladsavelyev replied:
JFYI, that's what hailctl sets automatically when you start a cluster with `--vep`, and we need it here explicitly because we using a custom initialisation script.

KatalinaBobowik replied:
Ok, good to know, thanks! @lgruen is it ok if I add this back in?

lgruen replied:
Thanks, @vladsavelyev, and sorry for misleading you, @KatalinaBobowik -- please add it back in.

vladsavelyev replied:
I've got a comment about these workers resources in the VEP setup instructions here: https://github.com/populationgenomics/production-pipelines/tree/from-initial/scripts/vep (this wasn't merged into main though because of change requests which I didn't get back to :-( that's why it's better to merge quickly even if the code is not ideal). I'll be moving these instructions into a new repo for managing scripts for reference data!

lgruen commented on scripts/vep/run_vep.py:
```diff
@@ -0,0 +1,34 @@
+    mt = hl.read_matrix_table(mt)
+    # filter to biallelic loci only
+    mt = mt.filter_rows(hl.len(mt.alleles) == 2)
+    mt = mt.filter_rows(mt.alleles[1] != '*')
```
Weren't we planning on skipping this filter? And I think there were a few other suggestions in the other PR.

KatalinaBobowik replied:
I don't think VEP works with starred alleles: https://discuss.hail.is/t/vep-output-variant-not-found-in-original-variants/1148

lgruen replied:
Ah, thanks. Could you add a comment explaining that, please?

lgruen commented on scripts/vep/run_vep.py:
```diff
@@ -0,0 +1,34 @@
+    mt = mt.filter_rows(mt.alleles[1] != '*')
+    vep = hl.vep(mt, config='file:///vep_data/vep-gcloud.json')
+    vep = vep.rows()
+    vep_path = output_path('vep105_GRCh38.mt')
```
Didn't we want to use VEP 104 for TOB-WGS?

KatalinaBobowik replied:
Is this ready to go now? If so I can change it

lgruen replied:
Can you check with @vladsavelyev about this, please? He's actively working on that.

Commit:
Run VEP on rows only; change version number and memory requirements

Commit:
Remove line about taking Matt's code, since there is no live link to this

KatalinaBobowik commented:
@vladsavelyev do you know if the `gs://cpg-reference/vep/vep-GRCh38.sh` file in the `dataproc.hail_dataproc_job` init parameter will change once your PR goes through?

Commit:
Explicitly set --vep cluster parameters

As per Vlad's comment: https://github.com/populationgenomics/tob-wgs/pull/117#discussion_r958020687

Commit:
Add in Vlad's changes after building VEP v104

lgruen commented on scripts/vep/run_vep.py:
```diff
@@ -0,0 +1,35 @@
+    ht = mt.rows()
+    # filter to biallelic loci only
+    ht = ht.filter(hl.len(ht.alleles) == 2)
+    ht = ht.filter(ht.alleles[1] != '*')
```
Can you add a comment explaining this filter (and maybe a link to the forum post), please?

KatalinaBobowik replied:
added

lgruen commented on scripts/vep/main.py:
```diff
@@ -0,0 +1,48 @@
+        worker_machine_type='n1-highmem-8',
+        worker_boot_disk_size=200,
+        secondary_worker_boot_disk_size=200,
+        script=f'{script} --mt {mt}',
```
Need to pass `vep_version` here?

vladsavelyev committed:
Fix

vladsavelyev committed:
Pass vep_version

vladsavelyev commented:
Running VEP 104.3 in tob-wgs-test: https://batch.hail.populationgenomics.org.au/batches/180699

Commit:
add comment about why starred alleles are removed

lgruen reviewed (approved):
Nicely done!
